source/simulator/solver.cc

```diff
@@ -499,25 +501,34 @@ namespace aspect
         // We hardcode the blocks down below, so make sure block 0 is indeed
         // the block containing velocity and pressure:
         Assert(introspection.block_indices.velocities == 0, ExcNotImplemented());
-        Assert(introspection.block_indices.pressure == 0, ExcNotImplemented());
+        Assert(introspection.block_indices.pressure == 0
+               ||
+               (parameters.include_melt_transport
+                && introspection.variable("fluid pressure").block_index == 0
+                && introspection.variable("compaction pressure").block_index == 0)
```

Comment:
Because they are in the same block...
@tjhei, I guess you can explain this better than I can.